portafly/src/components/pages/product/CreateProductPage.tsx

```diff
@@ -0,0 +1,182 @@
+import React, {
+  useState, FormEventHandler, FocusEventHandler, useEffect
+} from 'react'
+
+import {
+  PageSection,
+  TextContent,
+  Text,
+  Card,
+  CardBody,
+  Form,
+  FormGroup,
+  TextInput,
+  TextArea,
+  ActionGroup,
+  Button
+} from '@patternfly/react-core'
+import { useTranslation } from 'i18n/useTranslation'
+import { useHistory, Redirect } from 'react-router'
+import { createProduct, NewProduct } from 'dal/products'
+import { useAsync } from 'react-async'
+import { useAlertsContext } from 'components/util'
+import { ValidationException } from 'utils'
+
+type Validations = Record<string, {
+  validation: 'default' | 'success' | 'error',
+  errors?: string[]
+}>
+
+const CreateProductPage = () => {
+  // @ts-ignore missing strings file
+  const { t } = useTranslation('createProduct')
+  const { goBack } = useHistory()
+  const { addAlert } = useAlertsContext()
+
+  const [name, setName] = useState('')
+  const [systemName, setSystemName] = useState('')
+  const [description, setDescription] = useState('')
+  const [validations, setValidations] = useState<Validations>({
+    name: { validation: 'default' },
+    system_name: { validation: 'default' },
+    description: { validation: 'default' }
+  })
+
+  const {
+    isPending, error, run, data
+  } = useAsync({ deferFn: createProduct })
+
+  useEffect(() => {
+    if (error) {
+      if (Object.prototype.hasOwnProperty.call(error, 'validationErrors')) {
+        const { validationErrors } = (error as unknown as ValidationException)
+        const newValidations: Validations = {}
+        Object.keys(validationErrors).forEach((id) => {
+          newValidations[id] = {
+            validation: 'error',
+            errors: validationErrors[id]
+          }
+        })
+        setValidations({ ...validations, ...newValidations })
+      } else {
+        addAlert({ id: String(Date.now()), title: error.message, variant: 'danger' })
+      }
+    }
+  }, [error])
+
+  if (data) {
+    const { service } = data as NewProduct
+    return <Redirect to={`/products/${service.id}`} />
+  }
+
+  const isValid = validations.name.validation === 'success' && validations.system_name.validation !== 'error'
+
+  const validate = (inputName: string) => {
+    switch (inputName) {
+      case 'name':
+        return name.length > 5 ? 'success' : 'error'
+      case 'system_name':
+        // eslint-disable-next-line no-nested-ternary
+        return !systemName ? 'default' : (systemName.length > 5 ? 'success' : 'error')
+      case 'description':
+        return description.length > 0 ? 'success' : 'default'
+      default:
+        return 'default'
+    }
+  }
```

Comment:
We could have a similar approach to our `ActionHandlers` and `reducers`, so we can define the case, here would be the input, and the fn to execute or the value to return.